### DISCUSSION TITLE
Upstream clusters are not getting reflected properly after making use of destinationrule in envoy_cluster_upstream_* metrics

### DISCUSSION THREAD

**Discussion by Indresh2410** [istio/istio#51365] [General]
We're exploring setting up dashboards for traffic going through an envoy cluster

Eg: envoy_cluster_upstream_cx_active

But when metrics are exposed, cluster_name is always set to xds-grpc instead of actual destinationrule spec.host

Verified our apps are working as expected.

Wondering if we are missing something here

Istio:
client version: 1.17.2
control plane version: 1.16.4
data plane version: 1.16.4

**Comment by howardjohn** [comment]
Istio filters out many of the default envoy metrics because they are very expensive and not necessarily useful in Istio/Kubernetes context. For envoy_cluster_upstream_cx_active we exclude all but xds-grpc cluster.

See https://istio.io/latest/docs/ops/configuration/telemetry/envoy-stats/

**Comment by smktpd** [comment]
@Indresh2410 In the end, did you find any solution to visualize the number of active connections on per-host basis?
I've stumbled upon such task and looks like... Istio doesn't offer any solutions?
